src/Layout.js

```diff
@@ -528,16 +504,62 @@ var computeLayout = (function() {
       var/*float*/ totalFlexible = 0;
       var/*int*/ nonFlexibleChildrenCount = 0;
 
+      var/*int*/ firstFlex = -1;
+      var/*int*/ lastFlex = -1;
+
       var/*float*/ maxWidth;
-      for (i = startLine; i < node.children.length; ++i) {
+      for (i = startLine; i < childCount; ++i) {
         child = node.children[i];
+
+        // Pre-fill cross axis dimensions when the child is using stretch before
+        // we call the recursive layout pass
+        if (getAlignItem(node, child) === CSS_ALIGN_STRETCH &&
+            getPositionType(child) === CSS_POSITION_RELATIVE &&
+            isCrossDimDefined &&
+            !isDimDefined(child, crossAxis)) {
+          child.layout[dim[crossAxis]] = fmaxf(
+            boundAxis(child, crossAxis, node.layout[dim[crossAxis]] -
+              paddingAndBorderAxisCross -
+              getMarginAxis(child, crossAxis)),
+            // You never want to go smaller than padding
+            getPaddingAndBorderAxis(child, crossAxis)
+          );
+        } else if (getPositionType(child) === CSS_POSITION_ABSOLUTE) {
+          absoluteChildrenCount++;
```

Comment:
@kastiglione Generally, we try very hard to avoid allocations during layout as 1) allocations are not free (on Android, at least) 2) More allocations -> more likely GC calls and main thread stalls. It's a trade-off: more stable memory behaviour vs computational cost.